larpandoracontent/LArThreeDReco/LArHitCreation/HitCreationBaseTool.cc

```diff
@@ -122,11 +139,31 @@ void HitCreationBaseTool::GetBestPosition3D(const HitType hitType, const Cartesi
     LArGeometryHelper::MergeTwoPositions3D(this->GetPandora(), pCaloHit2D->GetHitType(), hitType, pCaloHit2D->GetPositionVector(),
         fitPosition, position3D, chi2);
 
+    double distanceToEdge = 0;
+    const double bestY(position3D.GetY());
+    const double bestZ(position3D.GetZ());
+
+    for (const LArTPCMap::value_type &mapEntry : larTPCMap)
+    {
+        const LArTPC* currentTPC = mapEntry.second;
+        // Check if the given hit is contained in the detector.
+        // If it is outside, then we get a positive value which
+        // will set the distanceToEdge value.
+        //
+        // We want to know the largest distance outside of the detector,
+        // so that we can make these 3D hits less favourable.
```

Comment:
I like having the comments here, but Pandora's standard line length is 140 chars, so would take advantage of the extra width to compress the comments a bit